src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -260,14 +298,19 @@ public Document GetOrAddExternalDocument(string filePath, string language)
             return new DocumentSpan(document, textSpan);
         }
 
-        private Document AddDocumentToProject(string filePath, string language, string projectName)
+        private Document? AddDocumentToProject(string filePath, string language, string projectName)
         {
-            var project = CurrentSolution.Projects.FirstOrDefault(p => p.Name == projectName && p.Language == language);
+            Project? project = CurrentSolution.Projects.FirstOrDefault(p => p.Name == projectName && p.Language == language);
             if (project == null)
             {
                 var projectInfo = ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Create(), projectName, projectName, language);
-                OnManagedProjectAdded(projectInfo);
+                OnProjectAdded(projectInfo);
                 project = CurrentSolution.GetProject(projectInfo.Id);
+
+                if (project == null)
```

Comment:
How could this happen? GetProject call can probably be replaced with GetRequiredProject so this can't bail here.